src/inmanta/protocol/openapi/converter.py

```diff
@@ -0,0 +1,185 @@
+        ][0]
+        return path
+
+    def extract_parameters_from_arg_options(self, response: Response, path: str, arg_options: Dict[str, ArgOption]):
+        parameters = []
+        for option_name, option in arg_options.items():
+            param_dict = {}
+            if option.header:
+                param_dict["in"] = ParameterType.header
+                param_dict["name"] = option.header
+                param_dict["schema"] = {"type": "string"}
+                if option.reply_header:
+                    response.headers = {option.header: {"description": option.header, "schema": {"type": "string"}}}
+            elif option_name in path:
+                param_dict["in"] = ParameterType.path
+                param_dict["name"] = option_name
+                param_dict["required"] = True
+                param_dict["schema"] = {"type": "string"}
+            if param_dict:
+                parameters.append(Parameter(**param_dict))
+        return parameters
+
+    def get_function_parameters(self, parameters: List[Parameter], url_method_function):
+        function_parameters = {
+            parameter_name: parameter_type
+            for parameter_name, parameter_type in url_method_function.__annotations__.items()
+            if parameter_name not in ["return", "tid"] and parameter_name not in [parameter.name for parameter in parameters]
+        }
+        return function_parameters
+
+    def convert_function_params_to_openapi(
+        self, function_parameters: Dict[str, Type], method_name: str, parameters: List[Parameter], path: str
+    ):
+        properties = {}
+        for parameter_name, parameter_type in function_parameters.items():
+            type_description = self.get_openapi_type(parameter_type)
+            if parameter_name not in path:
+                if method_name in ["PUT", "POST", "PATCH"]:
+                    properties[parameter_name] = type_description
+                else:
+                    parameters.append(
+                        Parameter(**{"name": parameter_name, "in": ParameterType.query, "schema": type_description})
+                    )
+            else:
+                parameters.append(
+                    Parameter(
+                        **{"name": parameter_name, "in": ParameterType.path, "required": True, "schema": type_description}
+                    )
+                )
+        return properties
+
+    def get_openapi_type(self, parameter_type: Type):
+        # TODO handle inmanta types
+        if hasattr(parameter_type, "__name__"):
+            type_name = parameter_type.__name__
+        else:
+            type_name = "dict"
+        python_to_openapi_types_without_format = {
+            "bool": "boolean",
+            "int": "integer",
+            "str": "string",
+            "dict": "object",
+            "list": "array",
+        }
+        openapi_type_name = python_to_openapi_types_without_format.get(type_name)
```

Comment:
1. why match on name, not on type or subtype
2. Wouldn't it be better to explicitly get pydantic types(which all have a common ancestor class) out (either to handle them or ignore them)